### DISCUSSION TITLE
Is there a way to get the onpress event for clicking on a google maps point of interest?

### DISCUSSION THREAD

**Discussion by jankytech** [react-native-maps/react-native-maps#4603] [Q&A] [answered]
Hello. I am not sure if this is a bug or a feature and what I would need to do to work around the issue I am having.

### Background

I am building a map based app, currently only on android but maybe in the future on iOS too.
I have a placecard popup that appears on top of the map when the user clicks a marker. The popup disappears when the user clicks the map.

 ### Problem

The above process works fine at the moment when the user clicks anywhere on the map that does not have a point of interest from google under the area they touched. If the user touches a point of interest instead of a plain part of the map, the onPress event from the MapView is not triggered and the code does not hit the logic that hides the placecard popup.

### Question

Is this intended behavior? If so, is there a way to hook into the onPress event that is being triggered instead of the MapView's event?

### Videos

In this video, I am clicking on the marker to bring up the placecard, then clicking on the map to hide the place card. All is working well here.
https://youtube.com/shorts/n-M5UHSed60

In this video, I am clicking on the marker to bring up the placecard, then clicking on a point of interest generated by google. In this example the placecard does not close because the onPress event from the MapView is not triggered
https://youtube.com/shorts/iS605zGzpxU

**Comment by lorenzomilesi** [comment, marked as answer]
There's another event to catch that click: `onPoiClick`
Take a look at [this comment](https://github.com/react-native-maps/react-native-maps/issues/3629#issuecomment-732450353)

**Reply by jankytech** [reply]
Thank you, that makes sense. I'll try it out next time I dig out the project.

I appreciate the reply
